### PR TITLE
Upgrade Knative serving to 0.11.1 

### PR DESCRIPTION
**Pull request by yuzisun**
**Which issue is resolved by this Pull Request:**
Resolves https://github.com/kubeflow/kfserving/issues/629

**Description of your changes:**
- taking knative 0.11.1 release and update kubeflow knative manifests, now minimum kube version is 1.14 and install all knative crds for v1alpha1, v1beta1, v1.
- removed knative gateway which is conflicting with kubeflow gateway
- change `config-istio` to point to kubeflow gateway
- change kfserving config to point to kubeflow gateway

**Checklist:**
- [ x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate`
    3. `make test`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/651)

### CODE REVIEW

**Comment by animeshsingh** [comment]
Thanks @yuzisun ! Putting this on hold for a while until we resolve knative 0.10 move in kfserving

/hold

File: knative/knative-serving-install/base/cluster-role.yaml [hunk 1]
```diff
@@ -118,6 +134,7 @@ rules:
   - admissionregistration.k8s.io
   resources:
   - mutatingwebhookconfigurations
+  - validatingwebhookconfigurations
```
Comment (krishnadurai) [inline knative/knative-serving-install/base/cluster-role.yaml]: The files for the webhook configurations seem to be missing.

File: knative/knative-serving-install/base/deployment.yaml [hunk 2]
```diff
@@ -74,12 +77,31 @@ spec:
       serviceAccountName: controller
       terminationGracePeriodSeconds: 300
 ---
+apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
```
Comment (krishnadurai) [inline knative/knative-serving-install/base/deployment.yaml]: Wouldn't it be better to have this in the same kustomize definition file hpa.yaml?

File: knative/knative-serving-install/base/cluster-role.yaml (hunk 1 above)
Comment (yuzisun) [reply knative/knative-serving-install/base/cluster-role.yaml]: the webhook configurations used to be generated, in 0.11.1 they moved to manifests.

File: knative/knative-serving-install/base/deployment.yaml (hunk 2 above)
Comment (yuzisun) [reply knative/knative-serving-install/base/deployment.yaml]: fixed.

File: knative/knative-serving-install/base/kustomization.yaml [hunk 3]
```diff
@@ -14,25 +14,25 @@ resources:
 - service.yaml
 - apiservice.yaml
 - image.yaml
-- hpa.yaml
```
[the remark this answers is not in the document]
Comment (yuzisun) [reply knative/knative-serving-install/base/kustomization.yaml]: fixed

**Comment by yuzisun** [comment]
following tests are failing..
E     kubeflow.testing.util.TimeoutError: Timeout waiting for deployment katib-manager in namespace kubeflow
E     kubeflow.testing.util.TimeoutError: Timeout waiting for deployment cloud-endpoints-controller in namespace kubeflow

**Comment by jlewi** [comment]
Here are a couple options

1. You can figure out why the tests are failing 

     * You can use stackdriver to track down the event logs for the deployments that timed out to 
        see what happened.

1. You can modify the test config to opt out of testing for knative 

     * Probably the easiest way to do that is  kubeflow/testing#585

**Pull request merged**